From 728e1920f47009d76c994a15f20ebac82305dc77 Mon Sep 17 00:00:00 2001
From: jagaur <jagaur@habana.ai>
Date: Wed, 04 Jun 2025 14:08:48 +0300
Subject: [PATCH] [SW-230770] gaudi3: fix crc32 calculation

This patch adds a WA to fix crc32 calculation error
which occurs when the file path contains a
hexadecimal string.

Change-Id: I08acbde6818566227d4bc7411b4e60a32b7e758c
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index bf10a81..6775843 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -145,7 +145,12 @@
 
 	# reverse CRC32 source as DMA engine works like that
 	hl_reverse_words ${FILE} ${FILE}.out 4
-	CRC32=$(crc32 ${FILE}.out)
+
+	# workround to avoid crc calculation error when absolute file path contains hex string
+	cp "${FILE}.out" /tmp/.
+	CRC32=$(crc32 /tmp/$(basename "${FILE}.out"))
+	rm -rf /tmp/$(basename "${FILE}.out")
+
 	echo "CRC32: ${CRC32}"
 	echo ${CRC32} | xxd -r -p > ${FILE}.crc32
 	hexdump -C ${FILE}.crc32
